### PR TITLE
Add options to extend list of sensitive keywords

### PR DESCRIPTION
**Pull request by Marloxo**
---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

this PR addressed the following issue: https://github.com/apache/airflow/issues/9372

basically it's adding support for additional labels from the configuration so users can extend the matching of sensitive keywords in variables name.

### CODE REVIEW

**Review by mik-laj — APPROVED** [review]
Can you add tests to avoid regression?

**Comment by Marloxo** [comment]
sure, I will take a look and update here

**Comment by mik-laj** [comment]
I would be happy if you would create a ticket in the Kubernetes library and add a link in the comment. Connection problems in this case should be handled on the library side.  If this is important to you, you can also ask for reviews of other Kubernetes contributors. This is open source, so every review is valuable.

**Comment by Marloxo** [comment]
@mik-laj Kubernetes library? I'm a bit lost here, I'm not using Kubernetes, it's just to hide variables value in Airflow UI 😅
not sure what Connection problems are you referring to, would you elaborate please

**Comment by mik-laj** [comment]
@Marloxo Sorry. I wanted to publish it in another PR.

**Comment by Marloxo** [comment]
@mik-laj no worries, any advice what tests I shall add ?

**Comment by mik-laj** [comment]
You should use tests.test_utils.config.conf_vars decorator and check different cases. I would be happy if you would use parameterized to easily check many cases.

**Comment by Marloxo** [comment]
@mik-laj I updated the PR with tests, would you please take a look.

File: airflow/config_templates/config.yml [hunk 1]
```diff
@@ -1678,6 +1678,13 @@
       type: string
       example: ~
       default: "True"
+    - name: sensitive_variable_fields
+      description: |
+        Sensitive keywords to look for in variables names
```
Comment (mik-laj) [inline airflow/config_templates/config.yml]:
```suggestion
        A comma-separated list of sensitive keywords to look for in variables names.
```

File: airflow/www/utils.py [hunk 2]
```diff
@@ -39,23 +39,28 @@
...
     if key_name:
-        config_set = conf.getboolean('admin',
-                                     'hide_sensitive_variable_fields')
-        field_comp = any(s in key_name.lower() for s in DEFAULT_SENSITIVE_VARIABLE_FIELDS)
+        config_set = conf.getboolean('admin', 'hide_sensitive_variable_fields')
+
+        sensitive_variable_fields = conf.get('admin', 'sensitive_variable_fields')
+
+        if sensitive_variable_fields:
+            DEFAULT_SENSITIVE_VARIABLE_FIELDS.update(sensitive_variable_fields.split(','))
```
Comment (mik-laj) [inline airflow/www/utils.py]: This will cause this variable to be updated each time this method is called. I think it's enough for this to happen only once.

**Comment by mik-laj** [comment]
Can you also add docs to https://airflow.readthedocs.io/en/latest/security.html?

File: airflow/www/utils.py (hunk 2 above)
Comment (Marloxo) [reply airflow/www/utils.py]: Good point

File: docs/security.rst [hunk 3]
```diff
@@ -421,3 +421,15 @@ the new key to the ``fernet_key`` setting, run
 #. Set ``fernet_key`` to ``new_fernet_key,old_fernet_key``
 #. Run ``airflow rotate_fernet_key`` to re-encrypt existing credentials with the new fernet key
 #. Set ``fernet_key`` to ``new_fernet_key``
+
+Sensitive Variable fields
+-------------------------
+
+By default, Airflow Value of a variable will be hidden if the key contains any words in
+(‘password’, ‘secret’, ‘passwd’, ‘authorization’, ‘api_key’, ‘apikey’, ‘access_token’), but can be configured
```
Comment (mik-laj) [inline docs/security.rst:429]: Can you add description for hide_sensitive_variable_fields option here also?

Comment (Marloxo) [reply docs/security.rst:429]: Sure, updated

**Review by mik-laj — APPROVED** [review]

**Comment by kaxil** [comment]
The tests were failing for this PR, it broke Master. Created a PR to fix it: https://github.com/apache/airflow/pull/9446
